### PR TITLE
Matrix Check Sum - initial check in

### PR DESCRIPTION
**Pull request by shrutivanw**
Initial check in for CS Fundamentals' assignment to check if the sum of values in a given row matches the sum of values in the corresponding column in the input matrix.
Refactored exercise 7 from older [Integers and Arrays](https://github.com/AdaGold/IntegersAndArrays) assignment.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,16 @@
-# matrix_check_sum
\ No newline at end of file
+# Matrix: Check sum
+Checks if the sum of values in a given row matches the sum of values in the corresponding column.
+
+## Exercise
+* Design and implement a method that checks that for the given matrix, where number of rows are equal to number of columns whether the sum of each row matches the sum of the corresponding column i.e. sum
+of numbers in row 'i' is the same as the sum of numbers in column 'i' for 'i' ranging from '0' to 'row.length-1'. If this is the case, the method return true. Otherwise, it return false.
+* For example, consider the input as:
```
Comment (Hamled) [inline README.md]: We need to use code blocks for this matrix to render properly in Markdown:
```
1 2 3 4
9 5 3 1
0 3 5 6
0 8 3 6
```

File: README.md [hunk 2]
```diff
@@ -1 +1,16 @@
...
+&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;1 2 3 4
+&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;9 5 3 1
+&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;0 3 5 6
+&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;0 8 3 6
+* In the above example case, the method should return true. (Sum of 0th row and 0th column is 10, sum of 1st row as well as the 1st column is 18 and so on.)
+* Share and explain the time and space complexities for your solution.
+* If you think of multiple approaches to solve the problem, implement the solution which minimizes space complexity. Explain the other approaches, and your decision in comments above the code.
+
+**Note**: Do not use Ruby provided functionality.
```
Comment (Hamled) [inline README.md]: What functionality are we talking about here?

I could definitely see this algorithm being something that they write using standard Ruby constructs such as `each` or other enumerable methods, and I'm not sure that would detract from their understanding.

I don't know if there are any Ruby libraries for dealing with matrices (I assume there are some Gems, not sure about in the standard lib), but that would be the most obvious thing to restrict them from using.

File: README.md [hunk 3]
```diff
@@ -1 +1,16 @@
-# matrix_check_sum
\ No newline at end of file
+# Matrix: Check sum
+Checks if the sum of values in a given row matches the sum of values in the corresponding column.
+
+## Exercise
+* Design and implement a method that checks that for the given matrix, where number of rows are equal to number of columns whether the sum of each row matches the sum of the corresponding column i.e. sum
```
Comment (Hamled) [inline README.md]: I know this isn't very helpful, but I feel like this explanation is difficult to understand. Could we split it out into multiple sentences?

**Review by Hamled — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (shrutivanw) [reply README.md]: Done.

File: README.md (hunk 2 above)
Comment (shrutivanw) [reply README.md]: Fair! I'll remove that line.

File: README.md (hunk 3 above)
Comment (shrutivanw) [reply README.md]: Now reworded.

**Review by Hamled — APPROVED** [review]

**Comment by Hamled** [comment]
Looks great!

**Pull request merged**